### PR TITLE
SOLR-10808: SchemaVersion 1.7 - DocValues default to true

### PR DESCRIPTION
**Pull request by HoustonPutman**
https://issues.apache.org/jira/browse/SOLR-10808

# Checklist

Please review the following and check all that apply:

- [ ] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

Need to:

- [x] Go through and update all of the test configSets
- [x] Make some tests for the schemaVersion logic
- [x] Confirm that we can do this in a minor release since the schemaVersion makes this backwards-compatible

### CODE REVIEW

File: solr/solr-ref-guide/modules/upgrade-notes/pages/major-changes-in-solr-9.adoc [hunk 1]
```diff
@@ -68,6 +68,18 @@ In Solr 8, it was possible to add docValues to a schema without re-indexing via
 Due to changes in Lucene 9, that isn't possible any more.
 
 == Solr 9.5
+
+=== SchemaVersion upgraded to 1.7
+Starting with Solr 9.5, the default schemaVersion is 1.7.
+With the new schema version, all fields and fieldTypes that support docValues (StrField, *PointField, BoolField, etc.)
+will have docValues enabled by default.
+
+In order to upgrade from schemaVersion 1.6 to 1.7, a re-index of all data will generally be required.
```
Comment (tflobbe) [inline solr/solr-ref-guide/modules/upgrade-notes/pages/major-changes-in-solr-9.adoc:77]: Should we mention that the re-index can't be in-place (for Solr 9 indices at least)?

**Review by tflobbe — APPROVED** [review]
+1!

File: solr/solr-ref-guide/modules/upgrade-notes/pages/major-changes-in-solr-9.adoc (hunk 1 above)
Comment (HoustonPutman) [reply solr/solr-ref-guide/modules/upgrade-notes/pages/major-changes-in-solr-9.adoc:77]: Why can't the re-index be in-place? Or do you mean atomic updates?

**Comment by tflobbe** [comment]
Also, should we consider making this change as part of this schema upgrade? https://issues.apache.org/jira/browse/SOLR-12963

File: solr/solr-ref-guide/modules/upgrade-notes/pages/major-changes-in-solr-9.adoc (hunk 1 above)
Comment (tflobbe) [reply solr/solr-ref-guide/modules/upgrade-notes/pages/major-changes-in-solr-9.adoc:77]: Lucene 9 enforces a strict schema. I believe adding docValues to an existing fields will not be allowed. See LUCENE-9334

**Review by risdenk — APPROVED** [review]

File: solr/solr-ref-guide/modules/upgrade-notes/pages/major-changes-in-solr-9.adoc (hunk 1 above)
Comment (HoustonPutman) [reply solr/solr-ref-guide/modules/upgrade-notes/pages/major-changes-in-solr-9.adoc:77]: Good catch, we should definitely mention that.

**Review by anshumg — APPROVED** [review]
This is gold!

File: solr/solr-ref-guide/modules/indexing-guide/pages/docvalues.adoc [hunk 2]
```diff
@@ -116,3 +119,13 @@ When retrieving stored fields, the insertion order is the return order.
 For docValues, it is the _sorted_ order.
 . For field types using `SORTED_SET` (see above), multiple identical entries are collapsed into a single value.
 Thus if values 4, 5, 2, 4, 1 are inserted, the values returned will be 1, 2, 4, 5.
+
+== Disabling DocValues
+
+DocValues are enabled by default for all field types that support them (when using a schemaVersion >= 1.7).
+Disabling docValues on a field or fieldType only requires adding `docValues="true"` to the definition:
```
Comment (anukanksha13) [inline solr/solr-ref-guide/modules/indexing-guide/pages/docvalues.adoc:126]: Shouldn't this be `docValues="false"` for disabling it on a field?

**Pull request merged**